app/controllers/author/submissions_controller.rb

```diff
@@ -123,6 +123,22 @@ def final_submission
     flash[:alert] = 'You are not allowed to visit that page at this time, please contact your administrator'
   end
 
+  def autorelease_extension
+    @submission = find_submission
+    status_giver = SubmissionStatusGiver.new(@submission)
+    status_giver.can_request_extension?
+
+    new_release_date = @submission.released_for_publication_at + 1.year
+    @submission.update!(released_for_publication_at: new_release_date)
+
+    redirect_to author_root_path
+    formatted_release_date = @submission.reload.released_for_publication_at.strftime("%B %d, %Y")
```

Comment:
I sometimes forget about `.reload`, so (golf clap) for making sure we actually are checking what we just saved to the db. 🎸